src/utils/hooks/transaction/hooks/use-fee-estimate.ts

```diff
@@ -0,0 +1,200 @@
+  'mutate' | 'mutateAsync'
+>;
+
+type UseFeeEstimateResult<T extends Transaction> = {
+  defaultCurrency: CurrencyExt;
+  currency: CurrencyExt;
+  selectProps: Pick<SelectProps, 'value' | 'onSelectionChange'>;
+  isEqualFeeCurrency: (currency: CurrencyExt) => boolean;
+} & ReactQueryUseFeeEstimateResult &
+  EstimateFunctions<T>;
+
+type FeeEstimateOptions = Omit<
+  UseMutationOptions<FeeEstimateResult, Error, EstimateFeeVariables, unknown>,
+  'mutationFn'
+>;
+
+const defaultFeeCurrency = GOVERNANCE_TOKEN;
+
+function useFeeEstimate<T extends Transaction>(type?: T, options?: FeeEstimateOptions): UseFeeEstimateResult<T> {
+  const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
+  const { data: pools } = useGetLiquidityPools();
+  const { getCurrencyFromTicker } = useGetCurrencies(bridgeLoaded);
+  const { getBalance } = useGetBalances();
+
+  const [feeCurrency, setFeeCurrency] = useState(defaultFeeCurrency);
+
+  const feeResultRef = useRef<FeeEstimateResult>();
+  const estimateFeeVariablesRef = useRef<EstimateFeeVariables>();
+
+  const mutateFee: MutationFunction<FeeEstimateResult, EstimateFeeVariables> = useCallback(
+    async ({ params, currency }) => {
+      const amount = await estimateTransactionFee(currency, pools || [], params);
+
+      const feeBalance = getBalance(currency.ticker)?.transferable;
+
+      const actionAmount = getActionAmount(params, amount.currency);
+
+      const isValid = isFeeValid(amount, feeBalance, actionAmount);
+
+      return {
+        amount,
+        isEqualToActionCurrency: !!actionAmount,
+        isValid
+      };
+    },
+    [getBalance, pools]
+  );
+
+  const { data, mutate, mutateAsync, ...feeMutation } = useMutation(
+    mutateFee,
+    mergeProps(options || {}, {
+      onSuccess: (data: FeeEstimateResult) => {
+        feeResultRef.current = data;
+      }
+    })
+  );
+
+  useErrorHandler(feeMutation.error);
+
+  const handleEstimateFee = useCallback(
+    (...args: Parameters<EstimateArgs<T>['estimate']>) => {
+      const params = getActionData(args, type);
+
+      const variables = { params, currency: feeCurrency };
+
+      estimateFeeVariablesRef.current = variables;
+
+      mutate(variables);
+    },
+    [type, feeCurrency, mutate]
+  );
+
+  const handleEstimateFeeAsync = useCallback(
+    (...args: Parameters<EstimateArgs<T>['estimate']>) => {
+      const params = getActionData(args, type);
+
+      const variables = { params, currency: feeCurrency };
+
+      estimateFeeVariablesRef.current = variables;
+
+      return mutateAsync(variables);
+    },
+    [type, feeCurrency, mutateAsync]
+  );
+
+  // Re-estimate fee based on latest stored variables
+  useInterval(() => {
+    if (!estimateFeeVariablesRef.current || feeMutation.isLoading) return;
+    mutate(estimateFeeVariablesRef.current);
+  }, REFETCH_INTERVAL.MINUTE);
+
+  const handleFeeSelectionChange = (ticker: Key) => {
```

Comment:
Shouldn't this be wrapped in `useCallback` to prevent unnecessary re-renders too?